From 1d145652e0c18a0451e5a56e64e8cc7a4e949b89 Mon Sep 17 00:00:00 2001
From: "Keith W. Campbell" <keithc@ca.ibm.com>
Date: Fri, 26 Apr 2024 14:09:19 -0400
Subject: [PATCH] Fix x86 assembly source issues with gcc 13

* remove ".arch" directive
* add .section .note.GNU-stack,"",@progbits at end of file

Signed-off-by: Keith W. Campbell <keithc@ca.ibm.com>
---
 runtime/oti/xhelpers.m4 | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/runtime/oti/xhelpers.m4 b/runtime/oti/xhelpers.m4
index 9d9fe46aba2..58f8294550e 100644
--- a/runtime/oti/xhelpers.m4
+++ b/runtime/oti/xhelpers.m4
@@ -93,7 +93,6 @@ define({SHORT_JMP},{short})
 
 define({FILE_START},{
 	.intel_syntax noprefix
-	.arch pentium4
 	.text
 })
 
@@ -111,7 +110,9 @@ define({START_PROC},{
 	GLOBAL_SYMBOL($1):
 })
 
-define({FILE_END})
+define({FILE_END},{
+	.section .note.GNU-stack,"",@progbits
+})
 
 define({END_PROC},{
 END_$1:
